### ISSUE TITLE
About the implementation in the code

### ISSUE THREAD

**Issue by wyp19930313** [jjery2243542/adaptive_voice_conversion#25]
Hello, I have two questions while reading your code. Could you please help me answer them if you are free.
1. Why is the forward propagation of training and prediction different?
`    def forward(self, x):   
        emb = self.speaker_encoder(x)      
        mu, log_sigma = self.content_encoder(x) 
        **_eps = log_sigma.new(*log_sigma.size()).normal_(0, 1)_**
        dec = self.decoder(mu + **_torch.exp(log_sigma / 2) * eps_**, emb)
        return mu, log_sigma, emb, dec

    def inference(self, x, x_cond):
        emb = self.speaker_encoder(x_cond)
        mu, _ = self.content_encoder(x)
        dec = self.decoder(mu, emb)
        return dec
`

2. How is the loss function of KL divergence calculated?
loss_kl = 0.5 * torch.mean(torch.exp(log_sigma) + mu ** 2 - 1 - log_sigma)
https://github.com/jjery2243542/adaptive_voice_conversion/blob/68c33518495d7de404a0f1fdce95e718db86c91b/solver.py#L86

**Comment by sbkim052** [comment]
hi, @wyp19930313 
I have the same questions as both of your questions. Have you found the answers?

**Comment by wyp19930313** [comment]
@sbkim052 
The first question is determined by the principle of VAE.
The second question is still not understood.
